models/src/test/scala/io/casperlabs/models/cltype/StandardCLSerializationTest.scala

```
@@ -0,0 +1,416 @@
+        CLType.FixedList(inner, length)
+
+      case key if key == "result_type" =>
+        val subData = getTable(data, key)
+        val ok      = parseCLType(getTable(subData, "ok_type"))
+        val err     = parseCLType(getTable(subData, "err_type"))
+        CLType.Result(ok, err)
+
+      case key if key == "map_type" =>
+        val subData = getTable(data, key)
+        val k       = parseCLType(getTable(subData, "key_type"))
+        val v       = parseCLType(getTable(subData, "value_type"))
+        CLType.Map(k, v)
+
+      case key if key == "tuple1_type" =>
+        val subData = getTable(data, key)
+        val t0      = parseCLType(getTable(subData, "_0"))
+        CLType.Tuple1(t0)
+
+      case key if key == "tuple2_type" =>
+        val subData = getTable(data, key)
+        val t0      = parseCLType(getTable(subData, "_0"))
+        val t1      = parseCLType(getTable(subData, "_1"))
+        CLType.Tuple2(t0, t1)
+
+      case key if key == "tuple3_type" =>
+        val subData = getTable(data, key)
+        val t0      = parseCLType(getTable(subData, "_0"))
+        val t1      = parseCLType(getTable(subData, "_1"))
+        val t2      = parseCLType(getTable(subData, "_2"))
+        CLType.Tuple3(t0, t1, t2)
+    }
+
+    def parseURef(data: Tbl): URef = {
+      val address = readHex(getString(data, "address"))
+      val access  = parseAccessRights(getString(data, "access_rights"))
+      URef(ByteArray32(address).get, access)
+    }
+
+    def parseAccessRights(s: String): AccessRights = s match {
+      case "READ"           => AccessRights.Read
+      case "WRITE"          => AccessRights.Write
+      case "READ_WRITE"     => AccessRights.ReadWrite
+      case "ADD"            => AccessRights.Add
+      case "READ_ADD"       => AccessRights.ReadAdd
+      case "ADD_WRITE"      => AccessRights.AddWrite
+      case "READ_ADD_WRITE" => AccessRights.ReadAddWrite
+      case _                => AccessRights.None
```

Wouldn't it be better to throw an error?

---

Done in https://github.com/CasperLabs/CasperLabs/pull/1782/commits/a108f0bc66369943950a419a9404887ea2c8abcc

---

Awesome. Thanks.